testing/regress/ecl/regex_find_fixed.ecl

```diff
@@ -0,0 +1,112 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2024 HPCC Systems®.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#OPTION('globalFold', FALSE);
+
+//------------------------------------------
+
+inDS := DATASET(['Colorless green ideas sleep furiously.'], {STRING s});
+
+// UTF-8 not included because the concept of a fixed-length
+// UTF-8 string does not make sense
+
+// buffer_x:    replacement occurs entirely within target buffer
+// alloc_x:     replacement requires extra temp buffer
+
+ResLayout := RECORD
+    STRING      unbounded_s;
+    STRING10    bounded_s;
+    UNICODE     unbounded_u;
+    UNICODE10   bounded_u;
+END;
+
+//------------------------------------------
+
+// Search for a word at beginning of string, return entire match
+STRING first_word_0_ps := '^\\w+' : STORED('first_word_0_ps');
+UNICODE first_word_0_pu := u'^\\p{L}+' : STORED('first_word_0_pu');
+
+first_word_0 := PROJECT
+    (
+        NOFOLD(inDS),
+        TRANSFORM
+            (
+                ResLayout,
+                SELF.unbounded_s := NOFOLD(REGEXFIND(first_word_0_ps, (STRING)LEFT.s, 0)),
```

Comment:
This test case doesn't actually change with your code enabled.  The same search is assigned to variable and fixed length targets, which means that the variable length result is reused.

Reply:
I pushed a change where all of the unbounded assignments were removed. That was the only way I could find to (easily) get the code generator to use the fixed-length versions.